tests/e2e/e2e_test.go

```diff
@@ -0,0 +1,153 @@
+package e2e
+
+import (
+	"context"
+	"crypto"
+	"fmt"
+	"math/rand"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/oasisprotocol/oasis-core/go/common/crypto/drbg"
+	"github.com/oasisprotocol/oasis-core/go/common/crypto/mathrand"
+	"github.com/oasisprotocol/oasis-core/go/common/crypto/signature"
+	memorySigner "github.com/oasisprotocol/oasis-core/go/common/crypto/signature/signers/memory"
+	oasisGrpc "github.com/oasisprotocol/oasis-core/go/common/grpc"
+	"github.com/oasisprotocol/oasis-core/go/common/quantity"
+	"github.com/oasisprotocol/oasis-core/go/oasis-node/cmd/common"
+	"github.com/stretchr/testify/require"
+
+	"github.com/oasisprotocol/oasis-core/go/common/entity"
+	consensus "github.com/oasisprotocol/oasis-core/go/consensus/api"
+	"github.com/oasisprotocol/oasis-core/go/consensus/api/transaction"
+
+	"github.com/oasisprotocol/oasis-core/go/oasis-node/cmd/debug/txsource/workload"
+	staking "github.com/oasisprotocol/oasis-core/go/staking/api"
+	v1 "github.com/oasisprotocol/oasis-indexer/api/v1"
+	"github.com/oasisprotocol/oasis-indexer/tests"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+const (
+	fundAccountAmount = 10000000000
+	timeout           = 1 * time.Second
+)
+
+func TestIndexer(t *testing.T) {
+	if _, ok := os.LookupEnv("OASIS_INDEXER_E2E"); !ok {
+		t.Skip("skipping test since e2e tests are not enabled")
+	}
```

Comment:
This block is copy-pasted >20 times. Can you extract it into a global helper please? `SkipTestIfEndToEndDisabled(t)` or `SkipUnlessE2E(t)` or `EnsureE2E(t)` or something.